scripts/add-icon-data.js

```diff
@@ -1,187 +1,132 @@
 import process from 'node:process';
-import chalk from 'chalk';
-import { input, confirm, checkbox } from '@inquirer/prompts';
-import autocomplete from 'inquirer-autocomplete-standalone';
-import getRelativeLuminance from 'get-relative-luminance';
+import { string, multiselect, select, toggle, log } from 'prask';
 import {
-  URL_REGEX,
   collator,
   getIconsDataString,
   titleToSlug,
   normalizeColor,
 } from '../sdk.mjs';
 import { getJsonSchemaData, writeIconsData } from './utils.js';
 
-const hexPattern = /^#?[a-f0-9]{3,8}$/i;
-
 const iconsData = JSON.parse(await getIconsDataString());
 const jsonSchema = await getJsonSchemaData();
 
-const titleValidator = (text) => {
-  if (!text) return 'This field is required';
-  if (
-    iconsData.icons.find(
-      (x) => x.title === text || titleToSlug(x.title) === titleToSlug(text),
-    )
-  )
-    return 'This icon title or slug already exist';
-  return true;
-};
-
-const hexValidator = (text) =>
-  hexPattern.test(text) || 'This should be a valid hex code';
+const aliasTypes = ['aka', 'old'].map((key) => ({
+  title: `${key} (${jsonSchema.definitions.brand.properties.aliases.properties[key].description})`,
+  value: key,
+}));
 
-const sourceValidator = (text) =>
-  URL_REGEX.test(text) || 'This should be a secure URL';
-
-const hexTransformer = (text) => {
-  const color = normalizeColor(text);
-  const luminance = hexPattern.test(text)
-    ? getRelativeLuminance.default(`#${color}`)
-    : -1;
-  if (luminance === -1) return text.toUpperCase();
-  return chalk.bgHex(`#${color}`).hex(luminance < 0.4 ? '#fff' : '#000')(
-    text.toUpperCase(),
+const licenseTypes =
+  jsonSchema.definitions.brand.properties.license.oneOf[0].properties.type.enum.map(
+    (license) => ({ title: license, value: license }),
   );
-};
 
-const aliasesTransformer = (text) =>
-  text
-    .split(',')
-    .map((x) => chalk.cyan(x))
-    .join(',');
-
-const aliasesChoices = Object.entries(
-  jsonSchema.definitions.brand.properties.aliases.properties,
-)
-  .filter(([k]) => ['aka', 'old'].includes(k))
-  .map(([k, v]) => ({ name: `${k}: ${v.description}`, value: k }));
+const isSecureURL = (input) => {
+  if (!URL.canParse(input)) {
+    return 'Must be a valid URL.';
+  }
+  if (new URL(input).protocol !== 'https:') {
+    return 'Must be a secure URL (HTTPS).';
+  }
+  return true;
```

Comment:
Note that this must be consistent with the rest of the simple-icons project. See https://github.com/simple-icons/simple-icons/blob/dcb07310d9f1db14a5dd33491346e2b197927377/sdk.mjs#L36-L39 and search for `URL_REGEX`.

Reply:
Darn. I saw that, but why use regex when you can use standard browser APIs and have finer control?

Reply:
Maybe you can update your `urlRegex` function over in https://github.com/simple-icons/simple-icons/pull/10276/files#diff-995497d12cba57a3d4c41acdd8c3503d4fb9a5155f617b9e3b2624a47ab48f1f instead?

Reply:
Yes, yes, you are right. I was too lazy on that implementation to check that was not breaking something 😅 Just commented https://github.com/simple-icons/simple-icons/pull/10276#issuecomment-1977571772 to not forget it.

Reply:
It would basically just be:

```js
const isValidURL = (input) => URL.canParse(input);
const isSecureURL = (input) => new URL(input).protocol === 'https:';
```

And then we can factor those into the validation of the URLs here.

Reply:
It isn't complete, check:

```js
> const input = 'https:a://hello/foo.bar'
> URL.canParse(input)
true
> new URL(input).protocol === 'https:'
true
> /^https:\/\/[^\s"']+$/.test(input)
false
```

That's why we are using the regex, but can be removed using string manipulation though must be checked if it is really faster.